xmtp_db/migrations/2025-04-22-121314_groups_add_fork_status/up.sql

```diff
@@ -0,0 +1,4 @@
+ALTER TABLE groups
+    ADD COLUMN fork_state BOOLEAN DEFAULT FALSE;
```

Comment:
Given this is a boolean rather than a blob or text, would `maybe_forked` be better?

Reply:
Also, given that we have a `DEFAULT` specified, could we add a `NOT NULL` constraint here, and make the resulting Rust code a simple `bool` rather than an `Option<bool>`? Should simplify the surrounding code.

I think we should consider doing a similar thing with `fork_details` too (`NOT NULL DEFAULT ''`), so that future Rust developers don't need to figure out whether the empty state is NULL or empty string

Reply:
Thanks for the comment, code is updated!